src/GraphQL/Instrumentation/TelemetryProvider.cs

```diff
@@ -0,0 +1,135 @@
+#if NET5_0_OR_GREATER
```

Comment:
Support for old frameworks is generally "as long as is practical"; hence we have no reason to remove ifdefs for .NET 5.

We do not wish to add additional dependencies to the project, so all changes here are scoped to .NET 5+.  Since OpenTelemetry is rather new, it would seem likely that users would be on a new version of .NET anyway.  If a user wishes to use OpenTelemetry with older/other frameworks, they may copy and paste the necessary class into their own project.

Reply:
@Shane32 Why this entire class is scoped to NET5+ ?

Reply:
@sungam3r  Although the `Activity` class exists, the required methods do not exist in prior versions. `ActivitySource` does not exist at all in prior versions.  See:

- https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.activity.settag?view=net-7.0
- https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.activitysource?view=net-7.0

As indicated by @Kielek , we can introduce a dependency on `System.Diagnostics.DiagnosticSource` to polyfill older versions.  However, this adds a dependency for GraphQL.NET which I do not believe we wish to do.

Second, OpenTelemetry is rather new (first 1.x version was released after .NET 5), and so support for OpenTelemetry is less likely to be used/needed by developers targeting older frameworks.

Third, the implementation is simple to copy-and-paste for developers wishing to use it in older frameworks, and considering the last couple points, not worth the trouble (in my opinion) to add a separate nuget package for this functionality which has been built into .NET for the last 2+ years.

Reply:
Understood. I agree.